src/pyenphase/updaters/meters.py

```diff
@@ -111,12 +132,73 @@ async def probe(
+
+        :param envoy_data: EnvoyData structure to store data to
+        """
+        # get the meter status and readings from the envoy
+        meters_status: list[CtMeterData] = await self._json_request(self.end_point)
+        meters_readings: list[dict[str, Any]] = await self._json_request(
+            self.data_end_point
+        )
+
+        envoy_data.raw[self.end_point] = meters_status
+        envoy_data.raw[self.data_end_point] = meters_readings
+
+        # pylint: disable=consider-using-enumerate
+        for index in range(len(meters_readings)):
+            meter = meters_readings[index]
+            # match meter identifier to one found during probe to identify production or consumption
+            if meter["eid"] == self.production_meter_eid and self.production_meter_type:
+                # if production meter was enabled (type known) store ctmeter production data
+                envoy_data.ctmeter_production = EnvoyMeterData.from_api(
+                    meter, meters_status[index]
+                )
+                # if more then 1 phase configured store ctmeter phase data
+                phase_production: dict[str, EnvoyMeterData] = {}
+                for phase in range(self.phase_count if self.phase_count > 1 else 0):
+                    production: EnvoyMeterData | None = EnvoyMeterData.from_phase(
+                        meter, meters_status[index], phase
+                    )
+                    # exclude None phases that were expected but not actually in report
+                    if production:
+                        phase_production[PHASENAMES[phase]] = production
+
+                if len(phase_production) > 0:
+                    envoy_data.ctmeter_production_phases = phase_production
+
+            # match meter identifier to one found during probe to identify production or consumption
+            if (
+                meter["eid"] == self.consumption_meter_eid
+                and self.consumption_meter_type
+            ):
+                # if consumption meter was enabled (type known) store ctmeter consumption data
+                envoy_data.ctmeter_consumption = EnvoyMeterData.from_api(
+                    meter, meters_status[index]
+                )
+
+                # if more then 1 phase configured store ctmeter pahse data
+                phase_consumption: dict[str, EnvoyMeterData] = {}
+                for phase in range(self.phase_count if self.phase_count > 1 else 0):
+                    consumption: EnvoyMeterData | None = EnvoyMeterData.from_phase(
+                        meter, meters_status[index], phase
+                    )
+                    # exclude None phases that were expected but not actually in report
+                    if consumption:
+                        phase_consumption[PHASENAMES[phase]] = consumption
+
+                if len(phase_consumption) > 0:
+                    envoy_data.ctmeter_consumption_phases = phase_consumption
```

Comment:
```suggestion
        for index, meter in enumerate(meters_readings):
            # match meter identifier to one found during probe to identify production or consumption
            if meter["eid"] == self.production_meter_eid and self.production_meter_type:
                # if production meter was enabled (type known) store ctmeter production data
                envoy_data.ctmeter_production = EnvoyMeterData.from_api(
                    meter, meters_status[index]
                )
                # if more then 1 phase configured store ctmeter phase data
                envoy_data.ctmeter_production_phases = {
                    PHASENAMES[phase_idx]: production
                    for phase_idx in range(
                        self.phase_count if self.phase_count > 1 else 0
                    )
                    # exclude None phases that were expected but not actually in report
                    if (
                        production := EnvoyMeterData.from_phase(
                            meter, meters_status[index], phase_idx
                        )
                    )
                }
            # match meter identifier to one found during probe to identify production or consumption
            if (
                meter["eid"] == self.consumption_meter_eid
                and self.consumption_meter_type
            ):
                # if consumption meter was enabled (type known) store ctmeter consumption data
                envoy_data.ctmeter_consumption = EnvoyMeterData.from_api(
                    meter, meters_status[index]
                )
                # if more then 1 phase configured store ctmeter phase data
                envoy_data.ctmeter_consumption_phases = {
                    PHASENAMES[phase_idx]: consumption
                    for phase_idx in range(
                        self.phase_count if self.phase_count > 1 else 0
                    )
                    # exclude None phases that were expected but not actually in report
                    if (
                        consumption := EnvoyMeterData.from_phase(
                            meter, meters_status[index], phase_idx
                        )
                    )
                }
```

Reply:
You'll likely have to adjust the tests as well for the empty dict instead of `None`

Reply:
or you could write it as 
```python
                phase_consumption = {
                    PHASENAMES[phase_idx]: consumption
                    for phase_idx in range(
                        self.phase_count if self.phase_count > 1 else 0
                    )
                    # exclude None phases that were expected but not actually in report
                    if (
                        consumption := EnvoyMeterData.from_phase(
                            meter, meters_status[index], phase_idx
                        )
                    )
                }

                if phase_consumption:
                    envoy_data.ctmeter_consumption_phases = phase_consumption
```

But might be easier to have an empty dict

Reply:
Thanks, I've implemented the `for` changes. Opted to return `none` to be consistent with [Production](https://github.com/pyenphase/pyenphase/blob/main/src/pyenphase/updaters/production.py) updater that does the same for the phases.